### PR TITLE
Fix query_timeout warning messages

### LINKED ISSUE #9801: False warn for "per-tenant timeout not configured"
**Describe the bug**
With Loki 2.8.2 simple scalable mode ( read and write targets ) per-tenant timeout is configured using runtime configuration. But Loki shows a warn `per-tenant timeout not configured, using default engine timeout (\"5m0s\"). This behavior will change in the next major to always use the default per-tenant timeout (\"5m\").`

**To Reproduce**
1. Create runtime configuration in runtime.yaml  
```
overrides:
      fake:
        query_timeout: 10m
```
2. Set runtime_config in loki config
```
runtime_config:
      file: /etc/loki_runtime/runtime.yaml
```
3. Start Loki of version 2.8.2
4. See the logs of read or write pods
5. The first line of log reports a warn like below
    

> level=warn ts=2023-06-26T14:24:12.394946274Z caller=loki.go:286 msg="per-tenant timeout not configured, using default engine timeout (\"5m0s\"). This behavior will change in the next major to always use the default per-tenant timeout (\"5m\")."

**Expected behavior**
Since `query_timeout` is already defined in runtime configuration this line of log is not expected to appear. Either Loki logging system is false or runtime configuration is not respected.

**Environment:**
 - gke
 - kustomize

**Screenshots, Promtail config, or terminal output**
image

### PR DESCRIPTION
**Pull request by aminesnow**
**What this PR does / why we need it**:
This PR fixes the warning messages that are logged in relation to the timeouts migration.

The messages reference the per-tenant config, but the config in question in the `AdjustForTimeoutsMigration` function is the global `limits_config`. This is a source of confusion for users.

**Which issue(s) this PR fixes**:
Fixes #9801

**Special notes for your reviewer**:
@owen-d @dannykopping 

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

**Comment by DylanGuedes** [comment]
Thank your for your contribution.

Could you elaborate a little more on what confused you? "Default per-tenant" and "global" are expected to mean the same thing, but "default per-tenant" fits better since it makes it explicit that once you define a timeout for a specific tenant, it won't be using the default anymore, while "global" might shadow that idea.

File: pkg/loki/loki.go [hunk 1]
```diff
@@ -280,35 +280,35 @@ func (c *Config) Validate() error {
...
 			),
 		)
 		return nil
 	}
 
-	if perTenantTimeoutIsDefault && !engineTimeoutIsDefault {
+	if globalTimeoutIsDefault && !engineTimeoutIsDefault {
 		if err := c.LimitsConfig.QueryTimeout.Set(c.Querier.Engine.Timeout.String()); err != nil {
-			return fmt.Errorf("couldn't set per-tenant query_timeout as the engine timeout value: %w", err)
+			return fmt.Errorf("couldn't set global query_timeout as the engine timeout value: %w", err)
```
Comment (DylanGuedes) [inline pkg/loki/loki.go:311]:
```suggestion
			return fmt.Errorf("couldn't set default per-tenant query_timeout as the engine timeout value: %w", err)
```

**Comment by aminesnow** [comment]
> Thank you for your contribution.
> 
> Could you elaborate a little more on what confused you? "Default per-tenant" and "global" are expected to mean the same thing, but "default per-tenant" fits better since it makes it explicit that once you define a timeout for a specific tenant, it won't be using the default anymore, while "global" might shadow that idea.

Hi @DylanGuedes :wave: 
The confusion comes when we define a per-tenant config in the runtime configuration and still get a warning saying that the per-tenant timeout is no configured (see #9801).

The `limits_config` documentation states:
> The limits_config block configures global and per-tenant limits in Loki.
That's why I think in this case using `global` instead of `per-tenant` clear that confusion.

**Comment by DylanGuedes** [comment]
> > Thank you for your contribution.
> > Could you elaborate a little more on what confused you? "Default per-tenant" and "global" are expected to mean the same thing, but "default per-tenant" fits better since it makes it explicit that once you define a timeout for a specific tenant, it won't be using the default anymore, while "global" might shadow that idea.
> 
> Hi @DylanGuedes 👋 The confusion comes when we define a per-tenant config in the runtime configuration and still get a warning saying that the per-tenant timeout is no configured (see #9801).
> 
> The `limits_config` documentation states:
> 
> > The limits_config block configures global and per-tenant limits in Loki.
> > That's why I think in this case using `global` instead of `per-tenant` clear that confusion.

Thank you, that explains it.

**Review by JoaoBraveCoding — APPROVED** [review]
lgtm, feels clearer

**Comment by aminesnow** [comment]
@DylanGuedes So what do you think about this change?

**Comment by DylanGuedes** [comment]
> @DylanGuedes So what do you think about this change?

Sorry for the back-and-forth; but could you modify the runtime docstrings to use the term "global" too?

**Comment by DylanGuedes** [comment]
> > > @DylanGuedes So what do you think about this change?
> > 
> > 
> > Sorry for the back-and-forth; but could you modify the runtime docstrings to use the term "global" too?
> 
> No worries @DylanGuedes ! Which docstrings is that exactly?

At `pkg/validation/limits.go`:
```
_ = l.QueryTimeout.Set(DefaultPerTenantQueryTimeout)
	f.Var(&l.QueryTimeout, "querier.query-timeout", "Timeout when querying backends (ingesters or storage) during the execution of a query request. If a specific per-tenant timeout is used, this timeout is ignored.")
```

**Review by DylanGuedes — APPROVED** [review]

**Comment by DylanGuedes** [comment]
> > pkg/validation/limits.go
> 
> I rephrased the docstring to make the distinction between per-tenant and global. WDYT?

Thanks for the patience! The CLA is pending, did you sign it? Once you sign it I'll gladly merge this one.

**Pull request merged**